app/controllers/placements/schools/mentors/add_mentor_controller.rb

```diff
@@ -0,0 +1,55 @@
+class Placements::Schools::Mentors::AddMentorController < Placements::ApplicationController
+  before_action :set_school
+  before_action :set_wizard
+
+  helper_method :step_path, :current_step_path, :back_link_path
+
+  def new
+    @wizard.reset_state
+    redirect_to step_path(@wizard.first_step)
+  end
+
+  def edit; end
+
+  def update
+    if !@wizard.save_step
+      render "edit"
+    elsif @wizard.next_step.present?
+      redirect_to step_path(@wizard.next_step)
+    else
+      @wizard.create_mentor
```

Comment:
```suggestion
      @wizard.create_mentor
      @wizard.reset_state
```
Don't forget to reset the state